pypeit/wavetilts.py

```diff
@@ -713,9 +774,108 @@ def run(self, doqa=True, debug=False, show=False):
                       'func2d':self.par['func2d'], 'nslit':self.slits.nslits,
                       'spat_order':self.spat_order, 'spec_order':self.spec_order,
                       'spat_id':self.slits.spat_id, 'bpmtilts': bpmtilts,
-                      'spat_flexure': self.spat_flexure, 'PYP_SPEC': self.spectrograph.name}
+                      'spat_flexure': self.spat_flexure, 'PYP_SPEC': self.spectrograph.name,
+                      'slits_calib_key': self.slits.master_key, 'tilt_traces': self.make_tbl_tilt_traces()}
         return WaveTilts(**tilts_dict)
 
+    def make_tbl_tilt_traces(self):
+        """
+        Make a Table of the  traced and fitted tilts from self.all_trace_dict
+        to save into the WaveTilts object
+
+        Returns:
+            `astropy.table.Table`_: Table including the traced and fitted tilts for each slit.
+            Columns are:
+
+                - slit_ids: Slit IDs for which the tilts were traced and fit
+                - goodpix_spat: Good spatial pixels of the traced tilts
+                - goodpix_tilt: Good spectral pixels  of the traced tilts
+                - goodpix_lid: Line IDs for each goodpix. This is needed to associate goodpix to each line
+                - badpix_spat: Masked spatial pixels of the traced tilts
+                - badpix_tilt: Masked spectral pixels of the traced tilts
+                - badpix_lid: Line IDs for each badpix. This is needed to associate badpix to each line
+                - good2dfit_spat: Good spatial pixels of the 2D fit of the tilts
+                - good2dfit_tilt: Good spectral pixels of the 2D fit of the tilts
+                - good2dfit_lid: Line IDs for each good2dfit. This is needed to associate good2dfit to each line
+                - bad2dfit_spat: Rejected spatial pixels of the 2D fit of the tilts
+                - bad2dfit_tilt: Rejected spectral pixels of the 2D fit of the tilts
+                - bad2dfit_lid: Line IDs for each bad2dfit. This is needed to associate bad2dfit to each line
+
+        """
+
+        if self.all_trace_dict is None:
+            msgs.error('No tilts have been traced and fit yet. Run the run() method first.')
+
+        # slit_ids
+        slit_ids = np.array([])
+
+        # good&bad traces
+        goodpix_lid = np.array([])
+        goodpix_spat = np.array([])
+        goodpix_tilt = np.array([])
+
+        badpix_lid = np.array([])
+        badpix_spat = np.array([])
+        badpix_tilt = np.array([])
+
+        # good&bad fits
+        good2dfit_lid = np.array([])
+        good2dfit_spat = np.array([])
+        good2dfit_tilt = np.array([])
+
+        bad2dfit_lid = np.array([])
+        bad2dfit_spat = np.array([])
+        bad2dfit_tilt = np.array([])
+
+        # fill the arrays with the traced and fitted tilts
+        # each trc is a slit
+        for i, trc in enumerate(self.all_trace_dict):
+            if trc is not None and self.slits.mask[i] == 0:
```

Comment:
I see a potential problem here where there is significant masking, namely the tilts are not going to be aligned with the slits in this table. I think it would be better to pre-allocate the table with a fixed size given by the number of slits. Then you can leave the masked slit fields just initialized to zero, and then check the mask in the plotting routines and only plot the slits that are not masked. It is also just better practice, since appending numpy arrays is very slow, i.e. it will involve copying things, whereas if this was done with lists of arrays, python would use pointers.